curl-helper.c

```diff
@@ -4670,7 +4670,9 @@ value caml_curl_multi_cleanup(value handle)
 
   caml_remove_generational_global_root(&h->values);
```

Comment:
and this makes me wonder if should remove root after cleanup because callbacks are tracked in h->values

Reply:
Shouldn't this line instead be moved after `caml_leave_blocking_section()` below?